templates/bond_Debian.j2

```diff
@@ -28,6 +28,26 @@ bond-xmit_hash_policy {{ item.bond_xmit_hash_policy }}
 {% endif %}
 {% if item.bond_slaves is defined and item.bond_mode == 'active-backup' %}
 bond-slaves none
+{% for slave in item.bond_slaves %}
+pre-up (sleep 2 && ifup {{ slave }}) &
+{% endfor %}
+{% endif %}
+{% if item.bond_slaves is defined and item.bond_mode == '802.3ad' %}
+bond-slaves {{ item.bond_slaves|join(' ') }}
+{% endif %}
+{% endif %}
+
+{% if item.bootproto == 'manual' %}
+iface {{ item.device }} inet manual
+{% if item.bond_mode is defined %}
+bond-mode {{ item.bond_mode }}
+{% endif %}
+bond-miimon {{ item.bond_miimon|default(100) }}
+{% if item.bond_slaves is defined and item.bond_mode == 'active-backup' %}
+bond-slaves none
+{% for slave in item.bond_slaves %}
+pre-up (sleep 2 && ifup {{ slave }}) &
```

Comment:
This seems quite brittle. It's run as a background task, so will likely start after the master has been brought up. It all seems a bit racy to me. Shouldn't the OS networking scripts handle these dependencies automatically? It seems to work on Ubuntu 16.04.

Reply:
I'm happy to remove this. I've seen issues where the bond master would not come up unless the slaves were up, but bringing up slaves first brings up the bond master. This may not be an issue with your configuration, so I'll take it out.